src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs

```diff
@@ -92,20 +103,31 @@ public void Dispose()
 
         public void Reset()
         {
-            if (_analyzerReference is AnalyzerFileReference analyzerFileReference)
+            ResetReferenceAndErrors(out var analyzerReference, out var analyzerLoadErrors);
+
+            if (analyzerReference is AnalyzerFileReference analyzerFileReference)
             {
                 analyzerFileReference.AnalyzerLoadFailed -= OnAnalyzerLoadError;
 
-                if (_analyzerLoadErrors != null && _analyzerLoadErrors.Count > 0)
+                if (!analyzerLoadErrors.IsEmpty)
                 {
                     _hostDiagnosticUpdateSource.ClearDiagnosticsForProject(_projectId, this);
                 }
 
                 _hostDiagnosticUpdateSource.ClearAnalyzerReferenceDiagnostics(analyzerFileReference, _language, _projectId);
             }
+        }
 
-            _analyzerLoadErrors = null;
-            _analyzerReference = null;
+        private void ResetReferenceAndErrors(out AnalyzerReference analyzerReference, out ImmutableArray<DiagnosticData> analyzerLoadErrors)
```

Comment:
Why the separate function? This seems to just make it harder to understand what Reset is doing because now I get to jump around.

Reply:
I like it that way since I dont need to look for _analyzerxxx in this method just to make sure it is not used.